### PR TITLE
Add delete wrapper method

### PR DESCRIPTION
**Pull request by sjohnston**
Does a little refactoring and then adds a wrapper for delete requests that adds the base to the url

### CODE REVIEW

File: lib/WWW/Mechanize.pm [hunk 1]
```diff
@@ -552,15 +540,43 @@ sub head {
...
     my $uri = shift;
 
+    $uri = $self->_uri_with_base( $uri );
+
+    # It appears we are returning a super-class method,
+    # but it in turn calls the request() method here in Mechanize
+    return $self->SUPER::head( $uri->as_string, @_ );
+}
+
+=head2 $mech->delete ($uri )
```
Comment (simbabque) [inline lib/WWW/Mechanize.pm:550]: Please indent this correctly. This is important in case we link to it from somewhere else in the documentation.

**Review by simbabque — CHANGES_REQUESTED** [review]
This is great, thank you very much. Could you fix that one typo in your POD please? You're welcome to amend and force-push your changes.

File: lib/WWW/Mechanize.pm (hunk 1 above)
Comment (oalders) [reply lib/WWW/Mechanize.pm:550]: Are you referring to the space? I'm seeing the same pattern at `=head2 $mech->head ($uri )
`

Comment (simbabque) [reply lib/WWW/Mechanize.pm:550]: Yes, I was. Hmm. I have been through this POD a few times to try to fix all of the inconsistencies, but have missed that. Ok, I think we take it anyway and I can do another pass after that. Thanks @oalders

**Comment by simbabque** [comment]
I see it wasn't your typo. Merged, thank you!

**Pull request merged**